### PR TITLE
migration(group): backfill substatus with reasonable defaults

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Apply default substatus values for current Groups.

### CODE REVIEW

File: src/sentry/migrations/0413_backfill_groupedmessage_substatus.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def backfill_substatus(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+
+    for group in RangeQuerySetWrapperWithProgressBar(
+        Group.objects.filter(status__in=(GroupStatus.UNRESOLVED, GroupStatus.IGNORED))
+    ):
+        if group.status == GroupStatus.UNRESOLVED and group.substatus is None:
+            group.substatus = GroupSubStatus.ONGOING
+        if group.status == GroupStatus.IGNORED and group.substatus is None:
+            group.substatus = GroupSubStatus.UNTIL_ESCALATING
```
Comment (wedamija) [inline src/sentry/migrations/0413_backfill_groupedmessage_substatus.py:19]: Need to `.save()` if any change

**Review by wedamija — COMMENTED** [review]
Since this is on the group table it might take a long time to run. If you want it to be a bit faster, you can follow the pattern here: https://github.com/getsentry/sentry/blob/74d267ebf81705866a5fe0e19ea3ddaf062952b3/src/sentry/migrations/0386_backfill_monitor_env_checkins.py#L1-L89

File: src/sentry/migrations/0413_backfill_groupedmessage_substatus.py (hunk 1 above)
Comment (barkbarkimashark) [reply src/sentry/migrations/0413_backfill_groupedmessage_substatus.py]: doh

**Review by wedamija — APPROVED** [review]

File: src/sentry/migrations/0417_backfill_groupedmessage_substatus.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    )
+
+    cursor = connection.cursor()
+    batch = []
+
+    for group_id, status, substatus in queryset:
+        if status == GroupStatus.UNRESOLVED and substatus is None:
+            batch.append((group_id, GroupSubStatus.ONGOING))
+        elif status == GroupStatus.IGNORED and substatus is None:
+            batch.append((group_id, GroupSubStatus.UNTIL_ESCALATING))
```
Comment (wedamija) [inline src/sentry/migrations/0417_backfill_groupedmessage_substatus.py:37]: Hmm, are all ignored issues ignored until escalating? Are we translating existing ignores over?

Comment (barkbarkimashark) [reply src/sentry/migrations/0417_backfill_groupedmessage_substatus.py:37]: For the time being, newly ignored issues will just be set to `UNTIL_ESCALATING`. We have some future work to replace `Group.IGNORED` -> `Group.ARCHIVED`. 

When we actually start implementing the logic to do the state transitions for Ignored to Unresolved(Escalating) or Unresolved(Ongoing), we can see about backfilling Ignored issues with the proper values.

Comment (armenzg) [reply src/sentry/migrations/0417_backfill_groupedmessage_substatus.py:37]: For context, there will be a follow up to undo this.

**Pull request merged**